qwerty_mlir/lib/Qwerty/IR/QwertyOps.cpp

```diff
@@ -322,8 +327,24 @@ struct CallIndirectIf : public mlir::OpRewritePattern<qwerty::CallIndirectOp> {
         rewriter.eraseOp(else_yield);
 
         rewriter.replaceOp(call, new_if_op.getResults());
-        // Now that the old scf.if has no uses, we can finally erase it
-        rewriter.eraseOp(if_op);
+        if (if_op.getResults()[0].use_empty()) {
+            rewriter.eraseOp(if_op);
+        } else {
+            mlir::Operation *tgt = nullptr;
+            for (mlir::Operation *user : if_op->getUsers()) {
+                mlir::Operation *op = user;
+                while (op->getBlock() != if_op->getBlock()) {
+                    op = op->getParentOp();
+                } 
+                if (!tgt) {
+                    tgt = op;
+                } else if (op->isBeforeInBlock(tgt)) {
+                    tgt = op;
+                }
+            }
+            assert(tgt);
```

Comment:
I think messages in your assertion are useful:
```suggestion
            assert(tgt && "No users of scf.if found, cannot move it");
```